dbaccess/block.go

```diff
@@ -160,99 +148,92 @@ func BluestBlock(ctx Context) (*dbmodels.Block, error) {
 // UpdateBlocksAcceptedByAcceptingBlock updates all blocks which are currently accepted by `currentAcceptingBlockID`
 // to be accepted by `newAcceptingBlockID`.
 // `newAcceptingBlockID` can be set nil.
-func UpdateBlocksAcceptedByAcceptingBlock(ctx Context, currentAcceptingBlockID uint64, newAcceptingBlockID *uint64) error {
-	db, err := ctx.db()
+func UpdateBlocksAcceptedByAcceptingBlock(ctx database.Context, currentAcceptingBlockID uint64, newAcceptingBlockID *uint64) error {
```

Comment:
Does `.Save()` in go-pg work the same silly way as in GORM?

If not, maybe it's a good idea to convert all/some .Update functions to .Save functions, to minimize code-complexity.

Reply:
You also didn't answer this one.

Reply:
I wrote on Telegram that I think this one should be done in a separate ticket. I can try to squeeze it tomorrow, but let's first merge this one, and make sure nothing is broken before doing any refactors.